### PR TITLE
Fix typos, improve error messages, etc

### PR DESCRIPTION
**Pull request by avag-sargsyan**
- Fixed typos
- Improved error messages
- Enforced dependency inversion in app service

### CODE REVIEW

File: internal/repositories/app.go [hunk 1]
```diff
@@ -28,8 +28,13 @@ func (ar AppRepository) IsExistDeploymentNameInDB(deploymentName string) (bool,
 	row := ar.db.QueryRow(query, deploymentName)
 	err := row.Scan(&id)
 	if err != nil {
-		ar.logger.Error("err while scan row in database ", err)
-		return false, err
+		if err == sql.ErrNoRows {
+			ar.logger.Error("no rows were returned ", err)
+			return false, err
```
Comment (siarhei-shliayonkin) [inline internal/repositories/app.go]: Is it an error? 
return false, nil - ?

File: internal/repositories/app.go [hunk 2]
```diff
@@ -39,7 +44,7 @@ func (ar AppRepository) IsExistDeploymentNameInDB(deploymentName string) (bool,
 func (ar AppRepository) SaveHashData(allHashData []*api.HashData, deploymentData *models.DeploymentData) error {
 	tx, err := ar.db.Begin()
 	if err != nil {
-		ar.logger.Error("err while saving data in database ", err)
+		ar.logger.Error("err while initiating a transaction to save data in database ", err)
```
Comment (siarhei-shliayonkin) [inline internal/repositories/app.go]: Maybe "open transaction error"? And logger.WithError(...)
Anyway, it seems that suggested correction is not correct. :)

File: internal/repositories/app.go (hunk 1 above)
Comment (avag-sargsyan) [reply internal/repositories/app.go]: > Is it an error? return false, nil - ?

Right, not an error.

File: internal/repositories/app.go (hunk 2 above)
Comment (avag-sargsyan) [reply internal/repositories/app.go]: > Maybe "open transaction error"? And logger.WithError(...) Anyway, it seems that suggested correction is not correct. :)

Ok, that seems better.

**Review by alexwatcher — APPROVED** [review]

File: internal/repositories/app.go [hunk 3]
```diff
@@ -28,8 +28,13 @@ func (ar AppRepository) IsExistDeploymentNameInDB(deploymentName string) (bool,
...
 	err := row.Scan(&id)
 	if err != nil {
-		ar.logger.Error("err while scan row in database ", err)
-		return false, err
+		if err == sql.ErrNoRows {
+			ar.logger.Error("no rows were returned ", err)
+			return false, err
+		} else {
+			ar.logger.Error("err while scan row in database ", err)
+			return false, err
```
Comment (siarhei-shliayonkin) [inline internal/repositories/app.go]: Warning, not Error

Comment (siarhei-shliayonkin) [reply internal/repositories/app.go]: or just skip this log

Comment (avag-sargsyan) [reply internal/repositories/app.go]: Right, I'll just add info.

**Pull request merged**